### ISSUE TITLE
Enterprise github repo: not generating events and unable to write commit status to github

### ISSUE THREAD

**Issue by sairamvoggu1007** [fluxcd/notification-controller#161]
Team,
version using : flux2 v0.8.0
events are not generating and unable to write commit status to github about the deployment

Following is error noticing in notification controller log.. It was supposed to fetch from my github domain but it's trying to fetch from api.github.com please check following error
 **```

{"level":"error","ts":"2021-03-12T18:40:29.577Z","logger":"event-server","msg":"failed to send notification","object":"platform/badlybehavingtestservice","kind":"Kustomizatio │
│ n","error":"could not list commit statuses: Get \"https://api.github.com/repos/s-voggu/eks/commits/d1cc91e17c740a4d203e8eef04b8312bab47f46b/statuses?per_page=50\": net/http:
```**


In gitrepository type resource I'm using ssh repo url and able to fetch latest commit status...In providers I'm using https repo url in address field.  In gitrespositry resource and kusomization resource events are getting generated but I don't see any events in alerts and providers . Observing above error in notification controller log please assist asap

**Comment by phillebaba** [comment]
I moved this issue to the NC so you do not need to create a new issue @sairamvoggu1007

**Comment by sairamvoggu1007** [comment]
Yes, please let me know once this is done I can test it.. currently we are on flux2 version0.8.0 @phillebaba

**Comment by simongottschlag** [comment]
Thank you @phillebaba for looking at it 😊🙏

**Comment by phillebaba** [comment]
@sairamvoggu1007 I created a draft PR to track any issues.
https://github.com/fluxcd/notification-controller/pull/162

Just checkout the branch and build+push the image to some image registry. Then replace the image used for notification-controller in your "fleet-infra" repository. I do not know if there are any major CRD changes between 0.8 and 0.9 that would complicate testing this, try and see if it works.

**Comment by sairamvoggu1007** [comment]
I will test it and let you know once it's done. thanks for the prompt response

**Comment by sairamvoggu1007** [comment]
@phillebaba we just built the image. I just need to replace the following image with the image I built right to test the changes?
```
              fieldPath: metadata.namespace
        image: ghcr.io/fluxcd/notification-controller:v0.9.0
        imagePullPolicy: IfNotPresent
```

**Comment by phillebaba** [comment]
Yep that is the one :)

**Comment by sairamvoggu1007** [comment]
@phillebaba unfortunately we don't have any environment open for testing so we can't test this from our side. Could you please let us know once this version is released. thanks

**Comment by phillebaba** [comment]
I cant really in good faith submit a PR and release something which I do not know works. You do not need a full cluster to test this, it would work if you install flux on a kind cluster locally on your computer which pulls from your GitHub instance. Could you do that?
